SteamKit2/SteamKit2/Steam/CDN/Client.cs

```diff
@@ -227,10 +232,31 @@ public async Task<int> DownloadDepotChunkAsync( uint depotId, DepotManifest.Chun
                 }
             }
 
+            await _lancacheDetectionLock.WaitAsync();
+            try
+            {
+                if ( !CheckedForLancacheServer )
+                {
+                    LancacheDetected = await LancacheDetector.DetectLancacheServerAsync(httpClient);
+                }
+            }
+            finally
+            {
+                CheckedForLancacheServer = true;
+                _lancacheDetectionLock.Release();
+            }
+
             var chunkID = Utils.EncodeHexString( chunk.ChunkID );
             var url = $"depot/{depotId}/chunk/{chunkID}";
 
-            using var request = new HttpRequestMessage( HttpMethod.Get, BuildCommand( server, url, cdnAuthToken, proxyServer ) );
+            var builtUrl = BuildCommand( server, url, cdnAuthToken, proxyServer );
+            using var request = new HttpRequestMessage( HttpMethod.Get, builtUrl );
+            if ( LancacheDetected )
+            {
+                request.Headers.Host = server.Host;
+                // User agent must match the Steam client in order for Lancache to correctly identify and cache Valve's CDN content
+                request.Headers.Add( "User-Agent", "Valve/Steam HTTP Client 1.0" );
```

Comment:
Do you mind linking me code in lancache that checks for this? I can't find it.

Reply:
seems like it's nginx that handles that
although it's in 2 places :thinking: 
https://github.com/lancachenet/monolithic/blob/d01c26ae0d92ba643017791a213573d0f54e0758/overlay/etc/nginx/conf.d/30_maps.conf#L2
https://github.com/lancachenet/monolithic/blob/d01c26ae0d92ba643017791a213573d0f54e0758/overlay/hooks/entrypoint-pre.d/15_generate_maps.sh#L27

Reply:
Okay, so we should be able to at least append (SteamKit) to the user agent here

Reply:
Tested appending earlier and it does not work. Instead an exact match is expected, which I believe is what the leading tilde signifies